tests/functional/lti_certification/test_lti_certification_1_1.py

```diff
@@ -0,0 +1,99 @@
+import time
+
+import oauthlib.common
+import oauthlib.oauth1
+import pytest
+from h_matchers import Any
+
+from lms.models import ApplicationInstance
+from tests.functional.base_class import TestBaseClass
+
+
+class TestLTICertification(TestBaseClass):
+    SHARED_SECRET = "TEST_SECRET"
+    OAUTH_CONSUMER_KEY = "Hypothesis1b40eafba184a131307049e01e9c147d"
+    OAUTH_NONCE = "38d6db30e395417659d068164ca95169"
+    OAUTH_CLIENT = oauthlib.oauth1.Client(OAUTH_CONSUMER_KEY, SHARED_SECRET)
+
+    def test_a_good_request_loads_fine(self, app, lti_params):
+        result = self.lti_launch(app, lti_params, status=200)
+
+        self.assert_response_is_html(result)
+
+        # Check a random parameter to see it's passed to the body
+        assert "tool_consumer_instance_guid" in result.text
+        assert lti_params["tool_consumer_instance_guid"] in result.text
+
+    def test_1_2_nice_message_when_res_link_id_and_return_url_missing(
+        self, app, lti_params
+    ):
+        lti_params.pop("resource_link_id")
+        lti_params.pop("launch_presentation_return_url")
+
+        result = self.lti_launch(app, lti_params, status=422)
+
+        self.assert_response_is_html(result)
+        assert "resource_link_id" in result
+
+    # ---------------------------------------------------------------------- #
+    # Assertions
```

Comment:
I think we can do without these divider comments. We always put all the helper methods, fixtures etc at the bottom of a class or module. And I think it's pretty clear that if something doesn't begin with `def test_` it's not a test, if it has `@pytest.fixture` it's a fixture, if it begins with `def assert_` it's for doing asserts, if none of the above it's just a helper

Reply:
I'll pluck them out in the next one, just helps my eyes keep things straight